src/Controller/AbstractController.php

```diff
@@ -109,13 +109,14 @@ public function dispatch(Request $request, Response $response = null)
         $this->response = $response;
 
         $e = $this->getEvent();
```

Comment:
Out of scope for this PR; I tried very hard only to update the code necessary to get the EM refactor working with it. If you feel strongly, open new PRs.